### PR TITLE
[palette_generator] Add PaletteGenerator.fromByteData

### PR DESCRIPTION
**Pull request by nt4f04uNd**
Fixes https://github.com/flutter/flutter/issues/28383

## Pre-launch Checklist

- [x] The title of the PR starts with the name of the package surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt.
- [x] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy].
- [x] I updated `CHANGELOG.md` to add a description of the change.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I signed the [CLA].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the `#hackers-new` channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning

### CODE REVIEW

**Comment by nt4f04uNd** [comment]
In discord @gspencergoog said
> As for the isolate, ideally it would be optional, but if it's easier to just always create it, then go ahead and do that.

I just checked and unfortunately currently spawn the isolate internally is not possible, because `filters` are closures, which cannot be sent over isolates, see https://github.com/dart-lang/sdk/issues/46623 . It is only possible that developer creates his own isolate

**Comment by gspencergoog** [comment]
Ahh, yes, that is true, I had forgotten about filters.

File: packages/palette_generator/lib/palette_generator.dart [hunk 1]
```diff
@@ -77,7 +98,17 @@ class PaletteGenerator with Diagnosticable {
     _selectSwatches();
   }
 
-  /// Create a [PaletteGenerator] from an [dart:ui.Image] asynchronously.
+  // TODO(gspencergoog): remove `dart:ui` paragragh from [fromByteData] method when https://github.com/flutter/flutter/issues/10647 is resolved
+
+  /// Create a [PaletteGenerator] asynchronously from encoded image [ByteData],
+  /// width and height. These parameters are packed in [EncodedImage].
```
Comment (gspencergoog) [inline packages/palette_generator/lib/palette_generator.dart:104]:
```suggestion
  /// width, and height. These parameters are packed in [EncodedImage].
```

**Review by gspencergoog — APPROVED** [review]
32384589-a60f0e74-c078-11e7-9bc1-e5b5287aea9d

**Pull request merged**